### DISCUSSION TITLE
Create df with the mortality raster using moving window

### DISCUSSION THREAD

**Discussion by cliu229** [andrewmarx/samc#22] [Q&A] [answered]
Hi Andrew and SAMC community, 

Thank you for developing this great package! 

I am getting some errors reported in the Model mortality data section. I replaced the road raster in the SAMC python example script with a nighttime light raster, but during the run I get the error _"cannot allocate vector of size 4320.9 Gb",_ but my raster file is only 245.14 KB, how can I fix the error during the moving window?

The code and warnings for the runtime error are as follows:

b_mort <- data.frame(
  roads = raster::extract(
    raster::focal(nightlight_zz, 
                  w = focalWeight(nightlight_zz, 990, 'rectangle'), 
                  fun = "sum", na.rm = T), b_points),
  mort = dplyr::select(b_points@data, mort = LIFESTATUS))

Error in h(simpleError(msg, call)) : 
cannot allocate vector of size 1687.2 Gb

Thankyou for reading this！

**Comment by andrewmarx** [comment, marked as answer]
My best guess is that the `990` in `focalWeight(nightlight_zz, 990, 'rectangle')` is way too big based on the unit size for the CRS of your raster. If you plot your raster and look at the axes, the `990` in your function is 990 times the distance of 1 unit on your axes. So first make sure your axes make sense, then pick a distance based on that.

**Reply by cliu229** [reply]
Thanks Andrew I am appreciated your quick reply， if I adjust the value to be smaller, for example, 10, the error message will become:
Error in `stop_subscript()`:
! Can't subset columns that don't exist.
x Column `LIFESTATUS` doesn't exist.

so I think it might be a problem with my raster data?

thankyou very much.

**Reply by cliu229** [reply]
When I run the label projection section, is there an error somewhere? 

proj4string <- "+proj=wgs +lat_0=34 +lon_0=114 +lat_1=34.1 +lat_2=34.6 +x_0=0 +y_0=0 +datum=WGS84 +units=m +no_defs" 
raster::crs(zz_land) <- proj4string 

I get the following:
Error in sp::CRS(value) : NA

**Reply by andrewmarx** [reply]
This is an issue with `dplyr::select(b_points@data, mort = LIFESTATUS)`. It's telling you that `b_points@data` (which I assume is a data frame), does not have a column called `LIFESTATUS`.

**Reply by cliu229** [reply]
Thanks for the guidance, I agree that this is really the problem.
